hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java

```diff
@@ -220,7 +222,7 @@ private int submitPurgeKeysRequest(List<DeleteBlockGroupResult> results,
 
     // Submit PurgeKeys request to OM
     try {
-      OzoneManagerRatisUtils.submitRequest(ozoneManager, omRequest, clientId, runCount.get());
+      OzoneManagerRatisUtils.submitRequest(ozoneManager, omRequest, clientId, callId.incrementAndGet());
```

Comment:
I see. So the problem is in each run there can be multiple unique key deletion requests, so it can't use runCount as call id otherwise only one will execute and the rest will be regarded as retry and skipped.